### PR TITLE
docs: update best practices regex groups + windows clrf + exporting test functions

### PR DESCRIPTION
**Pull request by PhilipAbed**
## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: docs/development/best-practices.md [hunk 1]
```diff
@@ -163,3 +164,14 @@ Use [JSDoc](https://jsdoc.app/index.html) to declare types and function prototyp
...
+
+### regex
+
+- use [Named Capturing Groups](https://www.regular-expressions.info/named.html) when capturing groups (e.g `(?<groupName>CapturedGroup`)
+
+### Windows
+
+- It's best if you set [core.autocrlf = input](https://git-scm.com/docs/gitattributes#_text) in your gitConfig,
+  otherwise the carriage return `\r\n` will confuse renovateBot in some cases.
+  it can also be done in your repository where you add `* text=auto eol=lf` to the `.gitattributes` file
```
Comment (HonkingGoose) [inline docs/development/best-practices.md:176]:
```suggestion
We recommend you set [`core.autocrlf = input`](https://git-scm.com/docs/gitattributes#_text) in your `gitConfig`, or the carriage return `\r\n` might confuse Renovate bot.
You can also set the line endings in your repository by adding `* text=auto eol=lf` to your `.gitattributes` file.
```

How about this?

File: docs/development/best-practices.md [hunk 2]
```diff
@@ -163,3 +164,14 @@ Use [JSDoc](https://jsdoc.app/index.html) to declare types and function prototyp
 - Omit constructors when defining Static classes
 - [No `#private` fields](https://google.github.io/styleguide/tsguide.html#private-fields). instead, use TypeScript's visibility annotations
 - Avoid underscore suffixes or prefixes, for example: `_prop`, use [whole words](https://google.github.io/styleguide/tsguide.html#properties-used-outside-of-class-lexical-scope) as suffix/prefix i.e. `internalProp`
+
+### regex
+
+- use [Named Capturing Groups](https://www.regular-expressions.info/named.html) when capturing groups (e.g `(?<groupName>CapturedGroup`)
```
Comment (HonkingGoose) [inline docs/development/best-practices.md:170]:
```suggestion
Use [Named Capturing Groups](https://www.regular-expressions.info/named.html) when capturing groups (e.g `(?<groupName>CapturedGroup`)
```

If there's one item, I don't think we should use a list. :wink: 

Are you sure about the example, it looks like you may be forgetting a `)` in the example string?

File: docs/development/best-practices.md [hunk 3]
```diff
@@ -134,6 +134,7 @@ Use `UTC` to be time zone independent.
 - Avoid `toMatchSnapshot`, only use it for:
   - huge strings like the Renovate PR body text
   - huge complex objects where you only need to test parts
+- Avoid exporting non-exported functions purely for the purpose of testing unless it's necessary
```
Comment (HonkingGoose) [inline docs/development/best-practices.md:137]:
```suggestion
- Avoid exporting functions purely for the purpose of testing unless you really need to
```

> Avoid exporting non-exported functions

This is strange to me. Would my suggestion say the same thing?

**Review by HonkingGoose — CHANGES_REQUESTED** [review]

File: docs/development/best-practices.md (hunk 3 above)
Comment (PhilipAbed) [reply docs/development/best-practices.md]: yes

File: docs/development/best-practices.md [hunk 4]
```diff
@@ -163,3 +164,13 @@ Use [JSDoc](https://jsdoc.app/index.html) to declare types and function prototyp
...
+
+### regex
+
+Use [Named Capturing Groups](https://www.regular-expressions.info/named.html) when capturing groups (e.g `(?<groupName>CapturedGroup`)
+
+### Windows
+
+We recommend you set [`core.autocrlf = input`](https://git-scm.com/docs/gitattributes#_text) in your `gitConfig`, or the carriage return `\r\n` might confuse Renovate bot.
+You can also set the line endings in your repository by adding `* text=auto eol=lf` to your `.gitattributes` file.
+-
```
Comment (HonkingGoose) [inline docs/development/best-practices.md:176]:
```suggestion

```

File: docs/development/best-practices.md (hunk 2 above)
Comment (PhilipAbed) [reply docs/development/best-practices.md]: i believe i wrote groups not group, but i added the word Multiple just to clear it out :D

**Review by viceice — DISMISSED** [review]

**Review by HonkingGoose — CHANGES_REQUESTED** [review]

**Review by viceice — APPROVED** [review]

**Review by HonkingGoose — APPROVED** [review]

**Pull request merged**